### DISCUSSION TITLE
resource management

### DISCUSSION THREAD

**Discussion by ac1dburnz** [Jip-Hop/jailmaker#203] [Q&A]
How do resources work here? If i wanted to create 10 jails, how much ram would be used compared to the 10 apps on the one jail?

**Comment by Jip-Hop** [comment]
I'd say try and measure it! 10 jails should use more ram than 1 jail but I don't think it will be a big impact.

**Comment by Lockszmith-GH** [comment]
tl;dr: Running 10 apps in separate namespaces should be very close to running those 10 apps in the same namespace, even if you introduce resource limiting, those limits would be per namespace, but they are not allocations (like a VM needs).

These are not VMs, they share the resources of the host OS.

Keep in mind that systemd-nspawn is [another interface to cgroups](https://en.wikipedia.org/wiki/Timeline_of_virtualization_development#2008) like Docker and podman.

From this (very good) [blog post](https://selectel.ru/blog/en/2017/02/09/systemd-containers-introduction-systemd-nspawn/):
> ### systemd-nspawn: General Information
> 
> The name systemd-nspawn is an abbreviation of _namespaces spawn_.
> From this name alone we can see that systemd-nspawn only manages isolated processes;
> it cannot isolate resources (however, this can be done with systemd, which we’ll talk about later on).
> These are just isolated processes that share certain resources based on how they are defined.
> Just like docker is a set of tooling to manage containerized packages, so are systemd-nspawn

**Reply by ac1dburnz** [reply]
Very very helpful! Thank you. I am trying to decide what I wanted to do for my app migration. A big disadvantage of k3s removal is that now I can no longer use metallb load balancer so I was trying to find a new solution so thank you :)

**Reply by Lockszmith-GH** [reply]
you can always spin a k3s cluster or [kind](https://kind.sigs.k8s.io) inside a jail :)

https://festivus.dev/kubernetes/
